manual/src/005_protocol-specification-rules.md

```diff
@@ -493,6 +493,43 @@ Thus, the complete example becomes:
 Note that the protocol description only specifies a model, but not which
 properties it might satisfy. We discuss these in the next section.
 
+### Clustering by Agent
+
+To enhance the clarity of the graph and identify the agents involved, nodes can be grouped by their respective agents. Clusters are purely a matter of display and do not affect the reasoning Tamarin uses to construct the proof. This is a purely aesthetic option, making the graph easier to read, which can be enabled or disabled by the user. The clusters will be displayed as colored boxes containing the relevant nodes. 
+
+The heuristic used for clustering identifies different sessions by checking if two nodes have the same agent and are connected by an edge; they will then be part of the same cluster. This method simplifies the organization of agents within the graph and ensures a clear representation of the agents and their respective nodes. The rules within a cluster will take the color of the cluster.
```

Comment:
The clustering splits different sessions of the same agent into different clusters. For that, it uses the following heuristic: two nodes from the same agent are part of the same cluster if they are connected by an edge resulting from the use of a linear fact. The rules within a cluster will take the color of the cluster.